packages/compass-e2e-tests/tests/connection.test.ts

```diff
@@ -705,8 +705,8 @@ describe('System CA access', function () {
       assertNotError(result);
       expect(result).to.have.property('ok', 1);
     } finally {
-      // make sure the browser gets closed otherwise if this fails the process wont exit
-      await afterTests(compass, this.currentTest, 'no-connect');
+      // make sure the browser gets closed otherwise if this fails the process won't exit
+      await cleanup(compass);
```

Comment:
We use two different patterns for how we manage these `cleanup`s in these changes. We use init in the test and `try.... finally -> cleanup` and then we also use the `cleanup` in the afterEach. Would it make sense to stick to one pattern there? I'm thinking folks when adding more tests might miss the need for having the `finally` and will bring over code, should we use the `cleanup` in the afterEach as a standard? Is there a nice way we can prevent the process not exiting without the future developer knowing this pattern?
Not a blocker

Reply:
It is supposed to just be before() with init and after() with cleanup but then people started doing init in tests, sometimes multiple times in the same test.

What is the pattern you would prefer?

Reply:
Not much you can do about the need for try / finally if you const compass = await init() inside a test, I think. Especially for the tests that do it multiple times.

I'm not really a fan of having let compass; in the describe() and then compass = await init() inside an it() and then cleanup(compass) in afterEach because the control/ownership of the compass variable becomes split all over the place. I think if we have to create compass inside a test I'd rather keep the var there because the afterEach can be who knows where in relation to the test and then you have to keep track of whether compass is undefined, an initialised compass var or one that was already cleanedup..

Reply:
I with you on the preferences order. Moving the init out of the test seems like a good pattern where we can.

Reply:
We usually do. I think the cases where we have it inside the test are usually because we have to pass different options to init() for each test and the tests are each testing a different option. For the final case I think the only tests that do that are the connection import/export ones and it kinda makes sense once you read through them. By definition those tests involve multiple compass sessions.